cmd/sandbox-ce/compose/compose.go

```diff
@@ -2,16 +2,135 @@ package compose
 
 import (
 	"context"
+	"fmt"
+	"io"
+	"net/http"
 	"os"
 	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/mitchellh/go-homedir"
+	"github.com/pkg/errors"
 )
 
-func Run(ctx context.Context, arg ...string) error {
-	cmd := exec.CommandContext(ctx, "docker-compose", arg...)
+// composeContainerPath is the url docker-compose is downloaded from in case
+// that it's not already present in the system
+const composeContainerPath = "https://github.com/docker/compose/releases/download/1.24.0/run.sh"
+
+var envKeys = map[string]bool{
+	"GITBASE_REPOS_DIR": true,
+}
+
+type Compose struct {
+	bin string
+}
+
+func (c *Compose) Run(ctx context.Context, arg ...string) error {
+	cmd := exec.CommandContext(ctx, c.bin, arg...)
+
+	var compOpts []string
+	for key := range envKeys {
+		val, ok := os.LookupEnv(key)
+		if !ok {
+			continue
+		}
+
+		compOpts = append(compOpts, fmt.Sprintf("-e %s=%s", key, val))
+	}
 
+	cmd.Env = []string{
```

Comment:
To make the command work in my linux machine, instead of replacing the `cmd` environment I changed this to append to a copy of the current process env, like this:
```go
cmd.Env = append(os.Environ(), fmt.Sprintf("COMPOSE_OPTIONS=%s", strings.Join(compOpts, " ")))
```

Reply:
Addressed [here](https://github.com/smacker/superset-compose/pull/4/commits/4956df0db7336fed033bf17058822f8ca96c4560).

Thanks for this! Today the same error that was occurring to you was happening to me, don't know why maybe I was always using the same shell with some exported env vars.